### PR TITLE
Move Get Invoices from Sales Billing button 

### PR DESCRIPTION
**Pull request by Saralrat**
I have moved the Get Invoices from Sales Billing button and reorganized the Tax Invoice page form.

image

Screenshot from 2024-11-11 00-25-22

### CODE REVIEW

**Comment by Saralrat** [comment]
image

File: sales_billing/constants.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+CUSTOM_FIELDS = {
+	"Payment Entry": [
```
Comment (kittiu) [inline sales_billing/constants.py]: please watch indentation, it is not alligned.

File: sales_billing/fixtures/custom_field.json [hunk 2]
```diff
@@ -1,214 +1,14 @@
...
-  "reqd": 0,
-  "search_index": 0,
-  "translatable": 0,
-  "unique": 0,
-  "width": null
- }
-]
\ No newline at end of file
+   {
+     "doctype": "Custom Field",
```
Comment (kittiu) [inline sales_billing/fixtures/custom_field.json]: We still use this file?

File: sales_billing/hooks.py [hunk 3]
```diff
@@ -8,15 +8,6 @@
 app_license = "MIT"
 
 fixtures = [
```
Comment (kittiu) [inline sales_billing/hooks.py]: Remove this line

File: sales_billing/public/js/payment_entry.js [hunk 4]
```diff
@@ -35,10 +35,6 @@ frappe.ui.form.on("Payment Entry", {
 			return;
 		}
 
-        var sales_billing = filters["sales_billing"];
-        frm.set_value("sales_billing", sales_billing);
-        if (!sales_billing) { return; }
```
Comment (kittiu) [inline sales_billing/public/js/payment_entry.js]: We don't need this? Why remove

**Comment by Saralrat** [comment]
Use sales_billing_dashboard.py to link Doctype.

image

**Comment by kittiu** [comment]
Looking good. But one last thing. Please add a patch that will rin on first time migrate.

**Comment by Saralrat** [comment]
> Looking good. But one last thing. Please add a patch that will rin on first time migrate.

Sorry, I forgot to add it. I've added it now.

File: sales_billing/constants.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+
+PROPERTY_SETTERS = {
+    "Withholding Tax Cert": [
+        (None, "default_print_format", "Withholding Tax Cert", "Data"),
+    ],
+    "Account": [
+        ("account_type", "options", "\nAccumulated Depreciation\nAsset Received But Not Billed\nBank\nCash\nChargeable\nCapital Work in Progress\nCost of Goods Sold\nDepreciation\nEquity\nExpense Account\nExpenses Included In Asset Valuation\nExpenses Included In Valuation\nFixed Asset\nIncome Account\nPayable\nReceivable\nRound Off\nStock\nStock Adjustment\nStock Received But Not Billed\nService Received But Not Billed\nTax\nTemporary\nCurrent Assets\nNon-current Assets\nCurrent Liabilities\nNon-current Liabilities", "Text"),
+    ],
```
Comment (kittiu) [inline sales_billing/constants.py:37]: Sorry, I think we don't need these lines.
Note: จริงๆพี่ก็งงอยู่ว่ามันติดมาได้ไง เพราะ Withholding Taxt Cert มันเป็นของ thai_tax สงสัยจะผิดมาตั้งแต่ต้น

File: sales_billing/constants.py [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+}
```
Comment (kittiu) [inline sales_billing/constants.py]:
```suggestion
```
Sorry, I think we don't need these lines.
Note: จริงๆพี่ก็งงอยู่ว่ามันติดมาได้ไง เพราะ Withholding Taxt Cert มันเป็นของ thai_tax สงสัยจะผิดมาตั้งแต่ต้น

File: sales_billing/hooks.py [hunk 7]
```diff
@@ -90,12 +68,12 @@
...
 
 # before_install = "sales_billing.install.before_install"
-# after_install = "sales_billing.install.after_install"
+after_install = "sales_billing.install.after_install"
 
 # Uninstallation
 # ------------
 
-# before_uninstall = "sales_billing.uninstall.before_uninstall"
+before_uninstall = "thai_payroll.setup.before_uninstall"
```
Comment (kittiu) [inline sales_billing/hooks.py]: Why are you referencing thai_payroll?
```suggestion
before_uninstall = "sales_billing.setup.before_uninstall"
```

File: sales_billing/constants.py (hunk 6 above)
Comment (Saralrat) [reply sales_billing/constants.py]: ก็คือไม่เอา PROPERTY_SETTERS เลยใช่มั้ยคะ

**Comment by kittiu** [comment]
So far, you should have learnt some coding concept. But the biggest concept we should hold is **Qulaity**. Please take time to review all changes and ensure they are near **Perfect** before submitting, ok?

**Comment by Saralrat** [comment]
> So far, you should have learnt some coding concept. But the biggest concept we should hold is **Qulaity**. Please take time to review all changes and ensure they are near **Perfect** before submitting, ok?

Okay, thank you. I will use it to improve myself.

**Comment by kittiu** [comment]
LGTM (look good to me now)

**Pull request merged**